server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -544,6 +554,11 @@ private IndexMetadata buildAndValidateTemporaryIndexMetadata(
         tmpImdBuilder.settings(indexSettings);
         tmpImdBuilder.system(isSystem);
 
+        if (remoteStoreBlobPathResolver != null) {
```

Comment:
should we also assert indexsettings contain remote store enabled? or is it not populated yet in the index settings?

Reply:
We are reusing the same remote attribute check that is used for putting replication type, remote enabled setting and other repository related settings.